### PR TITLE
separate default template for easy customization

### PR DESCRIPTION
**Pull request by superyyrrzz**
#662 
provide 2 functions for each type: preSteps() and postSteps, which defined in type.html.primary.extension.js. Users can change this file to modify the model.
@chenkennt @vwxyzh @hellosnow @ansyral @qinezh

### CODE REVIEW

**Comment by qinezh** [comment]
:shipit:

File: src/docfx.website.themes/default/ManagedReference.html.primary.extension.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// Copyright (c) Microsoft. All rights reserved. Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+exports.preSteps = function (model) {
+  return model;
```
Comment (hellosnow) [inline src/docfx.website.themes/default/ManagedReference.html.primary.extension.js]: question, how about process model directly and not return model itself?

Comment (superyyrrzz) [inline src/docfx.website.themes/default/ManagedReference.html.primary.extension.js]: As discussed, remain this as user may want to return a new model.

---
In reply to: [84629091](https://github.com/dotnet/docfx/pull/811#discussion_r84629091) [](ancestors = 84629091)

**Comment by hellosnow** [comment]
:shipit:

**Pull request merged**